WorkloadManagementSystem/DB/JobElasticDB.py

```diff
@@ -0,0 +1,128 @@
+""" DIRAC JobDB class is a front-end to the main WMS database containing
+  job definitions and status information. It is used in most of the WMS
+  components
+
+  The following methods are provided for public usage:
+
+  getJobParameters()
+
+  setJobParameter()
+"""
+
+__RCSID__ = "$Id$"
+
+from DIRAC.Core.Utilities.ReturnValues import S_OK, S_ERROR
+from DIRAC.Core.Base.ElasticDB import ElasticDB as DB
+from DIRAC.ResourceStatusSystem.Client.SiteStatus import SiteStatus
+
+#############################################################################
+
+JOB_STATES = ['Submitting', 'Received', 'Checking', 'Staging', 'Waiting', 'Matched',
+              'Running', 'Stalled', 'Done', 'Completed', 'Failed']
+JOB_FINAL_STATES = ['Done', 'Completed', 'Failed']
+
+
+class JobDB(DB):
+
+  def __init__(self):
+    """ Standard Constructor
+    """
+
+    DB.__init__(self, 'jobelasticdb', 'WorkloadManagement/JobElasticDB')
+
+    self.jobAttributeNames = []
+
+    self.siteClient = SiteStatus()
+
+    self.log.info("==================================================")
+
+#############################################################################
+  def getJobParameters(self, jobID, paramList=None):
+    """ Get Job Parameters defined for jobID.
+      Returns a dictionary with the Job Parameters.
+      If parameterList is empty - all the parameters are returned.
+    """
+
+    self.log.debug('JobDB.getParameters: Getting Parameters for job %s' % jobID)
+
+    resultDict = {}
+
+    if paramList:
+      paramNameList = []
+
+      for x in paramList:
+        paramNameList.append(x)
+
+      query = {
+          "query": {
+              "bool": {
+                  "must": [
+                      {"match": {"JobID": jobID}}, {"match": {"Name": ','.join(paramNameList)}}]}},
+          "_source": ["Name", "Value"]}
+
+    else:
+      query = {"query": {"match": {"JobID": jobID}}, "_source": ["Name", "Value"]}
+
+    result = self.query('jobelasticdb*', query)
+
+    if result['OK']:
+      if result['Value']:
+
+        sources = result['Value']['hits']['hits']
+
+        for i in xrange(0, len(sources)):
```

Comment:
The proper way to iterate over the entries in a list is:
```python
for source in sources:
  name = source['_source']['Name']
  ....
```
If the index is needed use the `enumerate` function (see docs)
( if sources is a dictionary with integer keys use `for index, source in sources.iteritems()` )

Reply:
I agree. Updated the same.